src/lib/rest/rest.cpp

```diff
@@ -807,72 +815,6 @@ bool urlCheck(ConnectionInfo* ciP, const std::string& url)
-   *  queryContext
-   *
-   * Note that in the case of unsubscribeContext and unsubscribeContextAvailability doesn't
-   * actually use a servicePath, but we also provide a default in that case for the sake of
-   * completeness
-   */
-
-  //
-  // FIXME P5: this strategy can be improved taking into account full URL. Otherwise, it is
-  // ambiguous (e.g. entity which id is "queryContext" and are using a conv op). However, it is
-  // highly improbable that the user uses entities which id (or type) match the name of a
-  // standard operation.
-  // Also, if we wait (if we can) until we know what service it is, this whole string-search will come for free
-  // Don't think strcasestr is a 'simple' function ...
-  //
-  if (strcasestr(url, "updateContextAvailabilitySubscription") != NULL) return DEFAULT_SERVICE_PATH_QUERIES;
-  if (strcasestr(url, "unsubscribeContextAvailability") != NULL)        return DEFAULT_SERVICE_PATH_QUERIES;
-  if (strcasestr(url, "subscribeContextAvailability") != NULL)          return DEFAULT_SERVICE_PATH_QUERIES;
-  if (strcasestr(url, "discoverContextAvailability") != NULL)           return DEFAULT_SERVICE_PATH_QUERIES;
-  if (strcasestr(url, "notifyContextAvailability") != NULL)             return DEFAULT_SERVICE_PATH_UPDATES;
-  if (strcasestr(url, "updateContextSubscription") != NULL)             return DEFAULT_SERVICE_PATH_QUERIES;
-  if (strcasestr(url, "unsubscribeContext") != NULL)                    return DEFAULT_SERVICE_PATH_QUERIES;
-  if (strcasestr(url, "subscribeContext") != NULL)                      return DEFAULT_SERVICE_PATH_QUERIES;
-  if (strcasestr(url, "registerContext") != NULL)                       return DEFAULT_SERVICE_PATH_UPDATES;
-  if (strcasestr(url, "updateContext") != NULL)                         return DEFAULT_SERVICE_PATH_UPDATES;
-  if (strcasestr(url, "notifyContext") != NULL)                         return DEFAULT_SERVICE_PATH_UPDATES;
-  if (strcasestr(url, "queryContext") != NULL)                          return DEFAULT_SERVICE_PATH_QUERIES;
-
-  /* Look for convenience operation. Subscription-related operations are special, all the other depend on
-   * the method
-   */
-  if (strcasestr(url, "contextAvailabilitySubscriptions") != NULL)      return DEFAULT_SERVICE_PATH_QUERIES;
-  if (strcasestr(url, "contextSubscriptions") != NULL)                  return DEFAULT_SERVICE_PATH_QUERIES;
-
-  if (strcasecmp(method, "POST")   == 0)                                return DEFAULT_SERVICE_PATH_UPDATES;
-  if (strcasecmp(method, "PUT")    == 0)                                return DEFAULT_SERVICE_PATH_UPDATES;
-  if (strcasecmp(method, "DELETE") == 0)                                return DEFAULT_SERVICE_PATH_UPDATES;
-  if (strcasecmp(method, "GET")    == 0)                                return DEFAULT_SERVICE_PATH_QUERIES;
-  if (strcasecmp(method, "PATCH")  == 0)                                return DEFAULT_SERVICE_PATH_UPDATES;
-
-  return DEFAULT_SERVICE_PATH_UPDATES;
```

Comment:
On a second thought, a CNR entry is needed. Note that with the defaultServicePath() approach POST /v2/subscription was incorrectly using "/" as its URL is not taken into account as a special case and the POST default applies.